src/ExpressionEvaluator/CSharp/Test/ResultProvider/TupleTests.cs

```diff
@@ -0,0 +1,530 @@
+                Assert.False(type.GetLmrType().IsTupleCompatible(out cardinality));
+                Assert.Equal(0, cardinality);
+
+                type = runtime.GetType("System.ValueTuple`2", typeof(int), typeof(string));
+                Assert.True(type.GetLmrType().IsTupleCompatible(out cardinality));
+                Assert.Equal(2, cardinality);
+
+                type = runtime.GetType("System.ValueTuple`3", typeof(int), typeof(string), typeof(int));
+                Assert.True(type.GetLmrType().IsTupleCompatible(out cardinality));
+                Assert.Equal(3, cardinality);
+
+                type = runtime.GetType("System.ValueTuple`4", typeof(int), typeof(string), typeof(int), typeof(string));
+                Assert.True(type.GetLmrType().IsTupleCompatible(out cardinality));
+                Assert.Equal(4, cardinality);
+
+                type = runtime.GetType("System.ValueTuple`5", typeof(int), typeof(string), typeof(int), typeof(string), typeof(int));
+                Assert.True(type.GetLmrType().IsTupleCompatible(out cardinality));
+                Assert.Equal(5, cardinality);
+
+                type = runtime.GetType("System.ValueTuple`6", typeof(int), typeof(string), typeof(int), typeof(string), typeof(int), typeof(string));
+                Assert.True(type.GetLmrType().IsTupleCompatible(out cardinality));
+                Assert.Equal(6, cardinality);
+
+                type = runtime.GetType("System.ValueTuple`7", typeof(int), typeof(string), typeof(int), typeof(string), typeof(int), typeof(string), typeof(int));
+                Assert.True(type.GetLmrType().IsTupleCompatible(out cardinality));
+                Assert.Equal(7, cardinality);
+
+                type = runtime.GetType("System.ValueTuple`8", typeof(int), typeof(string), typeof(int), typeof(string), typeof(int), typeof(string), typeof(int), typeof(string));
+                Assert.False(type.GetLmrType().IsTupleCompatible(out cardinality));
+                Assert.Equal(0, cardinality);
+
+                type = runtime.GetType("System.ValueTuple`1", typeof(string));
+                type = runtime.GetType("System.ValueTuple`8", typeof(int), typeof(string), typeof(int), typeof(string), typeof(int), typeof(string), typeof(int), ((TypeImpl)type.GetLmrType()).Type);
+                Assert.True(type.GetLmrType().IsTupleCompatible(out cardinality));
+                Assert.Equal(8, cardinality);
+
+                type = runtime.GetType("System.ValueTuple`9", typeof(int), typeof(string), typeof(int), typeof(string), typeof(int), typeof(string), typeof(int), typeof(string), typeof(int));
+                Assert.False(type.GetLmrType().IsTupleCompatible(out cardinality));
+                Assert.Equal(0, cardinality);
+            }
+        }
+
+        [Fact]
+        public void IsTupleCompatible_NonStruct()
+        {
+            var source =
+@"namespace System
+{
+    class ValueTuple<T1, T2> { }
```

Comment:
We have a workitem to block nonstruct ValueTuples, if possible. It is ok to have these tests, but do not stress too much the cases where ValueTuple is a class. It is nearly unsupported scenario already.